### ISSUE TITLE
Questions on reproduction details

### ISSUE THREAD

**Issue by fuzzall** [vsubramaniam851/multiagent-ft#3]
Thank the authors for sharing this interesting work. I recently met some difficulty in reproducing the results. I have some questions as follows.

1. `lm_ft.txt` cannot be directly used to set up the virtual environment, as it contains some local paths, which raises many errors. I tried to solve the conflicts by manually install those packages one by one but failed. Could you please check the environment and update `lm_ft.txt` accordingly?

1. The train/test split. As shown in these two lines, it seems that the results are obtained on the same set of samples, is it a typo?

https://github.com/vsubramaniam851/multiagent-ft/blob/6e61b7f0b1efe9a4729260cd2524cde24a40ab2b/multiagent-ft/original_gen.py#L174

https://github.com/vsubramaniam851/multiagent-ft/blob/6e61b7f0b1efe9a4729260cd2524cde24a40ab2b/multiagent-ft/ft_gen.py#L221

2. Since the link to the MATH dataset is expired, I tried to reproduce results on GSM8k with gpt-4o-mini through OpenAI API and Mistral-v0.2 served locally. Following the last question, I checked the performance of MAFT on the same set of samples (as in the codebase) and on another set of samples with no overlap with the training samples (as detailed in the paper). I obtained the following results:

![Image](https://github.com/user-attachments/assets/a2bc5407-96ee-4757-8956-3c133d2278b4)

Where "Inference w/o maft" denotes evaluating vanilla GPT/Mistral with MAD, and "Inference after maft" denotes evaluating MAFT-tuned GPT/Mistral with MAD. I record results on both hard_problems[:500] and hard_problems[500:1000]. 

Only MAFT on hard_problems[:500] achieved a slight improvement (91.6 to 92.6). While in the paper it was 75.6 to 85.6 on GPT-3.5-Turbo, it can be explained by the difference in the base model. However, no improvement was observed using Mistral or on another set of samples.

Would you mind clarifying these implementation details? Any information would be highly appreciated. Thank you!

**Comment by vsubramaniam851** [comment]
Hi,

Apologies for the delay!

1.	Noted – I will take a look! My sense is that the FastChat Finetuning API may be deprecated as well. 
2.	Thanks for catching this bug! Yes, the test problems should be held out. Fixed!
3.	There are a few details to keep in mind.
    a.	You will need to modify answer parsing in `ft_gen.py` and `ft_critic.py`, since these are designed for MATH but not for GSM. 
    b.	It’s important to be sure that you get a good improvement from multiagent debate from your model.
    c.	Note that gpt-3.5-turbo-0613 is still available for finetuning – see [here](https://platform.openai.com/docs/guides/fine-tuning). If you want to use gpt-4o-mini, this may need additional tuning. Similarly, the open-source models are sensitive and need quite a bit of tuning to stop overfitting. Be sure to tune your learning rate (it can vary depending on your data), weight decay and batch size, which were the most important factors to prevent overfitting.
